build.gradle.kts

```diff
@@ -0,0 +1,316 @@
+  // version" in case of master builds
+} else if (!file(".git").exists()) {
+  // To make sure it's safe for Docker image builds where .git folder is unavailable
+  version = "$PROSPECTIVE_RELEASE_VERSION-SNAPSHOT"
+} else {
+  val maybeSnapshot = if (ciBranch == "develop") "" else "-SNAPSHOT"
+
+  val git = org.ajoberstar.grgit.Grgit.open(mapOf("currentDir" to projectDir))
+  val lastTag = git.tag.list().sortedBy { it.dateTime }.last()
+  val commitsSinceLastTag =
+      git.log(mapOf("includes" to listOf("HEAD"), "excludes" to listOf(lastTag.fullName)))
+  val maybeCommitCount = if (commitsSinceLastTag.isEmpty()) "" else "-${commitsSinceLastTag.size}"
+  val shortCommitHash = git.head().abbreviatedId
+  val maybeDirty = if (git.status().isClean) "" else "-dirty"
+  git.close()
+
+  version =
+      "$PROSPECTIVE_RELEASE_VERSION$maybeCommitCount$maybeSnapshot+git.$shortCommitHash$maybeDirty"
+}
+
+dependencies {
+  // Checker is needed on root project runtime (not just compile-time) classpath for ArchUnit tests
+  testImplementation(rootProject.libs.checker.qual)
+  testImplementation(rootProject.libs.archunit)
+}
+
+apply<IntelliJPlugin>()
+
+configure<IntelliJPluginExtension> {
+  instrumentCode.set(false)
+  pluginName.set("git-machete-intellij-plugin")
+  version.set(intellijVersions["buildTarget"] as String)
+  plugins.set(listOf("git4idea")) // Needed solely for ArchUnit
+}
+
+if (!isCI) {
+  // The output of this task is for some reason very poorly cached,
+  // and the task takes a significant amount of time,
+  // while the index of searchable options is of little use for local development.
+  tasks.withType<BuildSearchableOptionsTask> { enabled = false }
+}
+
+tasks.withType<PatchPluginXmlTask> {
+  // `sinceBuild` is exclusive when we are using `*` in version but inclusive when without `*`
+  sinceBuild.set(
+      IntellijVersionHelper.toBuildNumber(intellijVersions["earliestSupportedMajor"] as String))
+
+  // In `untilBuild` situation is inverted: it's inclusive when using `*` but exclusive when without `*`
+  untilBuild.set(
+      "${IntellijVersionHelper.toBuildNumber(intellijVersions["latestSupportedMajor"] as String)}.*")
+
+  // Note that the first line of the description should be self-contained since it is placed into embeddable card:
+  // see e.g. https://plugins.jetbrains.com/search?search=git%20machete
+  pluginDescription.set(file("$rootDir/DESCRIPTION.html").readText())
+
+  changeNotes.set(
+      "<h3>v${rootProject.version}</h3>\n\n${file("$rootDir/CHANGE-NOTES.html").readText()}")
+}
+
+tasks.withType<RunIdeTask> { maxHeapSize = "4G" }
+
+tasks.withType<RunPluginVerifierTask> {
+  val maybeEap =
+      if (intellijVersions["eapOfLatestSupportedMajor"] != null)
+          listOf(
+              (intellijVersions["eapOfLatestSupportedMajor"] as String).replace(
+                  "-EAP-(CANDIDATE-)?SNAPSHOT".toRegex(), ""))
```

Comment:
Aaand another fix done by myself: `replace` extension method doesn't interpret the provided String as regex; an explicit conversion method `toRegex()` was needed here

Reply:
(this causes a string with `-EAP-SNAPSHOT` suffix to be passed to `ideVersions`, which crashed the entire `runPluginVerifier` task